subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt

```diff
@@ -1141,6 +1145,70 @@ class ProjectSchemaAccessorsIntegrationTest : AbstractPluginIntegrationTest() {
         }
     }
 
+    @Test
+    fun `can access project extension of nested type compiled to Java 11`() {
+
+        assumeJava11()
+
+        withFolders {
+            "buildSrc" {
+                "src/main/java/build" {
+                    withFile("Java11Plugin.java", """
+                        package build;
+
+                        import org.gradle.api.*;
+
+                        public class Java11Plugin implements Plugin<Project> {
+
+                            public static class Java11Extension {}
+
+                            @Override public void apply(Project project) {
+                                project.getExtensions().create("java11", Java11Extension.class);
+                            }
+                        }
+                    """)
+                }
+                withFile("settings.gradle.kts")
+                withFile("build.gradle.kts", """
+                    plugins {
+                        `java-library`
+                        `java-gradle-plugin`
+                    }
+
+                    java {
+                        sourceCompatibility = JavaVersion.VERSION_11
+                        targetCompatibility = JavaVersion.VERSION_11
+                    }
+
+                    gradlePlugin {
+                        plugins {
+                            register("java11") {
+                                id = "java11"
+                                implementationClass = "build.Java11Plugin"
+                            }
+                        }
+                    }
+                """)
+            }
+        }
+
+        withBuildScript("""
+            plugins { id("java11") }
+
+            java11 { println(this.javaClass.name) }
+        """)
+
+        assertThat(
+            build("-q").output,
+            containsString("build.Java11Plugin${'$'}Java11Extension")
+        )
+    }
+
+    private
+    fun assumeJava11() {
+        assumeTrue("Test requires Java 11", JavaVersion.current().isJava11Compatible)
+    }
```

Comment:
:o: I'd move this close to the existing assumptions here 

https://github.com/gradle/gradle/blob/611346be9533b5ffc4de03234991d9958449aa12/subprojects/kotlin-dsl-test-fixtures/src/main/kotlin/org/gradle/kotlin/dsl/fixtures/AbstractKotlinIntegrationTest.kt#L190-L198

Reply:
Done in 24dce2a.